exec.go

```diff
@@ -40,11 +40,24 @@ func (scope VarMap) SetWriter(name string, v SafeWriter) VarMap {
 	return scope
 }
 
+func (t *Template) Runtime() *Runtime {
+	st := pool_State.Get().(*Runtime)
```

Comment:
This will give you *any* runtime from the pool, possibly a new one, not the one executing the template.

Reply:
Instead, you should create a runtime, then create a custom resolver to which you pass the runtime, then set the resolver in the runtime, then execute the template.